### ISSUE TITLE
Book images are getting clipped

### ISSUE THREAD

**Issue by jschlade** [openaudible/openaudible#1302]
Using 3.9.6 on a Mac.

Book images are being clipped in the latest version of the app.

Here's an example:

<img width="1667" alt="Screenshot 2023-11-06 at 1 46 08 PM" src="https://github.com/openaudible/openaudible/assets/13022731/f09fd428-215e-4b83-9c8b-4a46a6476c1d">

**Comment by openaudible** [comment]
Hm. Unable to reproduce. Does resizing the window make it draw properly? Is the image ok in the Action: "Get Info" window for the image? 

Thanks for the report.. will try to add some code to monitor it.

**Comment by jschlade** [comment]
Looks like this only is an issue on Macs. I don't see this problem on Windows.

Have you tried on a Mac? Running macOS Sonoma version 14.1

**Comment by openaudible** [comment]
Yeah, upgraded to Sonoma here. We have a high-DPI monitor.. not sure if that would make a difference. 
If you resize the window does it remain clipped?

**Comment by jschlade** [comment]
Maybe you could try starting/stoping/starting the application a few times. This might reproduce it.

I have a dual monitor configuration. I have 2 [27" Dell UltraSharp U2718Q(https://www.displayspecifications.com/en/model/fed0d61)]

**Comment by openaudible** [comment]
I released a beta that adds some code to help debug this. 
https://openaudible.org/beta
If the problem still persists, go to About Menu: Diagnostics, and a web page will show a bunch of diagnostics.  
A new line called "main_window_check" should be present with info about the cover image.

I think this is a bug in the Eclipse SWT libraries we are using, so it may take just waiting for the next update.

A few more questions: 

Does switching between resolutions immediately show the problem or do you need to restart the app? I don't have access to 3360 x 1890 and wasn't able to see it on any resolution.. but didn't restart the app each time.

Are you on an Intel or Aarch64 processor?

Let me know if you try the beta...

**Comment by jschlade** [comment]
I'm on Aarch64. Is there a build for this?

**Comment by openaudible** [comment]
Yeah we have a universal binary so works well on both. But a problem on Intel may not show on in the aarch64 version and vv. So good to know when there is a tricky bug.

**Comment by openaudible** [comment]
Oh shoot. Let me figure that out.

**Comment by openaudible** [comment]
Ok. Sounds like that might have been a deadlock bug that can be tricky to diagnose. The new beta opens a log window and checks for deadlock every 20 seconds. Give it a try if you can.

**Comment by jschlade** [comment]
Okay this beta doesn't hang for me and seems to run fine. I've installed it and it's still clipping the image.

Is there something I can gather for you?

Also in general I find these standalone java apps interesting to follow like OpenAudible as I do some java development myself. One question for this team is: 

> Have you ever consider using JavaFX vs Eclipse SWT?
>  If so, what was the deciding factor to use Eclipse SWT?

Thanks

**Comment by openaudible** [comment]
Hmm. Sounds like an obscure gtk swt bug. 

May clear up when we get an updated swt library. 


we just picked swt over javafx. Eclipse makes a good bunch of products and it is all open source. 

All risk depreciation due to desktop apps becoming more niche. 

Sorry no ideas on how to help. 

Marking as unsolvable (by me). 😠
